cli/running/running.go

```diff
@@ -470,50 +460,65 @@ func FillCtx(cliOpts *config.CliOpts, cmdCtx *cmdcontext.CmdCtx,
 		instEnabledPath = basePath
 	}
 
-	instParams, err := collectInstances(appName, cliOpts, instEnabledPath)
-	if err != nil {
-		return fmt.Errorf("can't find an application init file: %s", err)
+	var appList []string
+	if len(args) == 0 {
+		appList, err = util.CollectAppList(instEnabledPath)
+		if err != nil {
+			return fmt.Errorf("can't collect an application list "+
+				"from instances enabled path: %s", err)
+		}
+	} else {
+		appList = append(appList, args[0])
 	}
 
-	// Cleanup instances list.
 	runningCtx.Instances = nil
-	for _, inst := range instParams {
-		var instance InstanceCtx
-		var runDir string
-		var logDir string
-		var dataDir string
-
-		instance.AppPath = inst.AppPath
-		instance.AppName = inst.AppName
-		instance.InstName = inst.InstName
-
-		if cliOpts.App != nil {
-			runDir = cliOpts.App.RunDir
-			logDir = cliOpts.App.LogDir
-			dataDir = cliOpts.App.DataDir
-			instance.LogMaxSize = cliOpts.App.LogMaxSize
-			instance.LogMaxAge = cliOpts.App.LogMaxAge
-			instance.LogMaxBackups = cliOpts.App.LogMaxBackups
-			instance.Restartable = cliOpts.App.Restartable
+	for _, appName := range appList {
+		if strings.HasSuffix(appName, ".lua") {
+			appName = appName[0 : len(appName)-4]
+		}
+		instParams, err := collectInstances(appName, cliOpts, instEnabledPath)
```

Comment:
Shouldn't this be under else of previous condition? Does it make sense to search for instances for app.lua?
What does "Params" in this var name mean?

Reply:
> Shouldn't this be under else of previous condition?

No, why?

> Does it make sense to search for instances for app.lua?

Yes it does, look at the implementation of this function. If an application is just a single file, there will be one instance.

> What does "Params" in this var name mean?

Sorry, the code and namings are literally not mine. Fixed.